### PR TITLE
PR #18: Implementation for issue #53

### LINKED ISSUE #53: Planning trajectory from given waypoints
The current planning trajectory looks different from the given waypoints, especially in the final phase of the vehicle, which prevents the crash happens in case 0, case 1, and case 2.
Refactor the `ac3r_plus.py` and update the function `generate_trajectory` to support the stop condition.

### PR DESCRIPTION
**Pull request by harveyvn**
In this PR, we have:
- Fixed a clockwise in planning trajectory.
- Added a `parking` segment to the `ac3rs` module beside `turn` and `straight` segments.
- Replaced the `ac3r_plus.py` with the `ac3rp` module.
- Updated the lasted JSON files for 7 cases generated from AC3R.
- Added a speed to generating_trajectory's points to support RoadProfiler.

### CODE REVIEW

File: python_src/models/ac3rp/common.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    # For us clockwise must be positive so we need to invert the sign
+    direction = copysign(1.0, cross(NORTH, B))
+    angle_between_segments = atan2(abs(cross(NORTH, B)), dot(NORTH, B))
+    #
+    intial_rotation = degrees(direction * angle_between_segments)
+
+    return initial_location, intial_rotation
+
+
+def f7(seq):
```
Comment (alessiogambi) [inline python_src/models/ac3rp/common.py]: What's this? Maybe you can give the method a more descriptive name.

File: python_src/models/ac3rp/segment.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def __init__(self):
+        self.type = "parking"
+        self.length = PARKING_CONSTRAINT
+
+
+class Straight:
+    def __init__(self, point_a: Point, point_b: Point):
+        # Straight segment - Note we care about length only at this point
+        # Rotation is implied by the previous elements or initial rotation
+        self.type = "parking"
```
Comment (alessiogambi) [inline python_src/models/ac3rp/segment.py]: This probably should be "straight"

**Review by alessiogambi — CHANGES_REQUESTED** [review]
Please address the two comments and resubmit.

File: python_src/models/ac3rp/common.py (hunk 1 above)
Comment (harveyvn) [reply python_src/models/ac3rp/common.py]: The method was updated! Many thanks 👍🏻

File: python_src/models/ac3rp/segment.py (hunk 2 above)
Comment (harveyvn) [reply python_src/models/ac3rp/segment.py]: Many thanks 👍🏻

**Review by alessiogambi — APPROVED** [review]
LGTM

**Pull request merged**